### PR TITLE
refactor(neon): make NeonPlatform an interface

### PR DESCRIPTION
**Pull request by Leptopoda**

### CODE REVIEW

File: packages/neon/neon/lib/src/platform/android.dart [hunk 1]
```diff
@@ -5,22 +5,44 @@ import 'package:path/path.dart' as p;
...
-              throw MissingPermissionException(Permission.storage);
-            }
-            return p.join((await getExternalStorageDirectory())!.path);
-          },
-        );
+class AndroidNeonPlatform implements NeonPlatform {
+  const AndroidNeonPlatform();
+
+  @override
+  bool get canUseCamera => true;
```
Comment (provokateurin) [inline packages/neon/neon/lib/src/platform/android.dart:14]:
```suggestion
  bool canUseCamera = true;
```

Works as well

**Comment by Leptopoda** [comment]
Looks like ci doesn't like PRs from a fork :)

**Review by provokateurin — CHANGES_REQUESTED** [review]

File: packages/neon/neon/lib/src/platform/android.dart (hunk 1 above)
Comment (Leptopoda) [reply packages/neon/neon/lib/src/platform/android.dart:14]: this then would also need to be made final and our linting doesn't like field initializes in const classes

Comment (provokateurin) [reply packages/neon/neon/lib/src/platform/android.dart:14]: Ahhh

**Review by provokateurin — APPROVED** [review]

**Comment by provokateurin** [comment]
Ah shit I think the conventional commits CI doesn't work for forks

**Comment by provokateurin** [comment]
Also your branch name is wrong

**Comment by provokateurin** [comment]
Reopened https://github.com/hyiso/commitlint/issues/7

**Comment by Leptopoda** [comment]
> Also your branch name is wrong

So the branch name will also be linted? So `refactor/neon_platform` or what?

**Comment by provokateurin** [comment]
No the branch name is not linted, I just noticed that the branch name is wrong for this PR

**Pull request closed without merging**